### DISCUSSION TITLE
Storage Progress Bar (supabase-js v2)

### DISCUSSION THREAD

**Discussion by HirotakaAseishi** [supabase/supabase#10210] [Questions]
Hi, 
 I want to use storage upload with progress bar. 
 So, How do I get the headers for XMLHttpRequest with supabase-js v2?
        
 (#23 's code may be old one now )

**Comment by danieldilly** [comment]
When a user logs in, the `onAuthStateChange` event fires with the `SIGNED_IN` event.  This event contains a `session` object which contains an `access_token`.

When you upload your file with `XMLHttpRequest`, you can simply set these headers:
```
{ key: 'apikey', value: access_token },
{ key: 'Authorization', value: `Bearer ${access_token}` }
```

**Comment by inian** [comment]
You can listen to progress events with our resumable uploads feature in [Storage v3](https://supabase.com/blog/storage-v3-resumable-uploads). Here are the docs on [resumable uploads](https://supabase.com/docs/guides/storage/uploads#resumable-upload).

**Reply by HirotakaAseishi** [reply]
Thank you so much! 　It is very convenient, including the updated UI😊
 (I'm sorry, I missed first answer)
